internal/api/dto/subscription.go

```diff
@@ -529,3 +529,131 @@ type SubscriptionUpdatePeriodResponseItem struct {
 	Success        bool      `json:"success"`
 	Error          string    `json:"error"`
 }
+
+// CreateSubscriptionLineItemVersionRequest represents a request to create new versions of subscription line items
+type CreateSubscriptionLineItemVersionRequest struct {
+	OldPriceVersionID string `json:"old_price_version_id" validate:"required"`
+	NewPriceVersionID string `json:"new_price_version_id" validate:"required"`
+}
+
+// Validate validates the subscription line item version creation request
+func (r *CreateSubscriptionLineItemVersionRequest) Validate() error {
+	if err := validator.ValidateRequest(r); err != nil {
+		return ierr.WithError(err).
+			WithHint("Invalid subscription line item version creation request").
+			Mark(ierr.ErrValidation)
+	}
+
+	// Validate that old and new price version IDs are different
+	if r.OldPriceVersionID == r.NewPriceVersionID {
+		return ierr.NewError("old_price_version_id and new_price_version_id must be different").
+			WithHint("Please provide different price version IDs for versioning").
+			Mark(ierr.ErrValidation)
+	}
+
+	return nil
+}
+
+// CreateSubscriptionLineItemVersionResponse represents the response for subscription line item version creation
+type CreateSubscriptionLineItemVersionResponse struct {
+	ProcessedCount int                                 `json:"processed_count"`
+	UpdatedItems   []SubscriptionLineItemVersionUpdate `json:"updated_items"`
+}
+
+// SubscriptionLineItemVersionUpdate represents a single line item version update
+type SubscriptionLineItemVersionUpdate struct {
+	SubscriptionID string                        `json:"subscription_id"`
+	OldLineItem    *SubscriptionLineItemResponse `json:"old_line_item"`
+	NewLineItem    *SubscriptionLineItemResponse `json:"new_line_item"`
+}
+
+// SyncPriceChangesToSubscriptionsRequest represents a request to sync price changes to all affected subscriptions
+type SyncPriceChangesToSubscriptionsRequest struct {
+	// PriceVersionMappings maps old price IDs to new price IDs for versioning
+	PriceVersionMappings []PriceVersionMapping `json:"price_version_mappings" validate:"required,min=1"`
+	// BatchSize is the number of subscriptions to process in each batch (default: 100)
+	BatchSize int `json:"batch_size,omitempty"`
+	// DryRun if true, will only simulate the changes without applying them
+	DryRun bool `json:"dry_run,omitempty"`
+}
+
+// PriceVersionMapping represents a mapping from old price to new price
+type PriceVersionMapping struct {
```

Comment:
are we expecting the caller to keep track of the old and new price ids - i.e keep track of the requested price id changes and then pick the new from response?